components/remote_settings/src/remote_settings.udl

```diff
@@ -73,3 +77,50 @@ interface RemoteSettings {
     [Throws=RemoteSettingsError]
     void download_attachment_to_path(string attachment_id, string path);
 };
+
+/// Stores application-level configuration data and constructs clients
+interface RemoteSettingsService {
+    constructor(RemoteSettingsConfig2 config);
+
+    /// Construct a new client with the configuration passed to the constructor
+    [Throws=RemoteSettingsError]
+    RemoteSettingsClient make_client(string collection_name);
+};
+
+/// Configuration for RemoteSettingsService
+///
+/// Once all clients have moved to the new API, we can remove the old RemoteSettingsConfig and
+/// rename this.
+dictionary RemoteSettingsConfig2 {
+    /// Server to use
+    RemoteSettingsServer server;
+    /// Directory to store data in.  Must be writable by the client.
+    string storage_dir;
+    /// Override bucket name ("main" by default, "preview" for preview collections)
+    string? bucket_name = null;
+};
+
+/// New remote settings client
+///
+/// This is a WIP matches the semantics of the official client more closely.  Once this is complete,
+/// we may decide to deprecate and remote the RemoteSettings class.
+interface RemoteSettingsClient {
+    /// Get the current set of records.
+    ///
+    /// If no records have been downloaded, this will return null.  Use `sync` to fetch the current
+    /// records from the server.
+    [Throws=RemoteSettingsError]
+    sequence<RemoteSettingsRecord>? get();
```

Comment:
Are we going to include the dumps in remote settings as well?

This would avoid the application having to load them separately, or pass them in. We can also include some of the logic that the desktop client uses to determine when the local records should be updated from the dump or not.

If so, then I'd expect this to be more `If no records have been downloaded, this will load from the local dumps, if found. Otherwise this will return null`.

Reply:
Yes, but I plan to add this later on.  I'm wondering if that should be a different class though, since if you know there's a local dump then `get()` doesn't need to return a nullable value.